catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -189,7 +189,7 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
     try {
       QueryResponse solrResponse;
 
-      if (solrFilterDelegate.isIdQuery()) {
+      if (filterAdapter.adapt(request.getQuery(), new RealTimeGetDelegate())) {
         LOGGER.debug("Performing real time query");
         SolrQuery realTimeQuery = getRealTimeQuery(query, solrFilterDelegate.getIds());
```

Comment:
I'm not convinced `getRealTimeQuery` forms the correct query for a query like `(ID = 123 AND foo = bar) OR (ID = 234 AND fizz = buzz)`. looks like it may erroneously turn that query to `ID = 123, 234 and fizz = buzz`

Reply:
maybe try it in a unit test

Reply:
I queried with this 
`[[[ id = 4e7d93a6a6bc463783d2b346fba0dfe6 ] AND [ resource-uri is like file:/C:/workspace/ddf/distribution/ddf/target/ddf-2.13.11-SNAPSHOT/data/cdm/00aeba4b6958410f9b14d4a86e488f28 ]] OR [[ id = 22af0559f867461b94971c3d0e5043e1 ] AND [ title is like INC ]]]`
which came out as expected and returned the expected results. I think getRealTimeQuery appends those Id's at the end because Solr is just going to fetch the entries by Id then apply the filter to what was retrieved.

Reply:
and it's smart enough not to muddle the conditions? it doesn't accidentally apply conditions on id 123 that were meant for id 234?

Reply:
It must get it right because otherwise that query wouldn't have given back any results. The filters ANDed with the id filters are both unique to the metacard with that id. If they were applied to both ids nothing should have been returned. I also did a query where one of those filters didn't match the metacard with the id and it only gave me the metacard with the correct filters.